### PR TITLE
ALS-1622: Adopt amf-custom-validator snapshot 16 and improve logging

### PR DESCRIPTION
**Pull request by agusayerza**

### CODE REVIEW

File: als-server/js/src/main/scala/org/mulesoft/als/server/modules/diagnostic/JsCustomValidator.scala [hunk 1]
```diff
@@ -4,15 +4,23 @@ import org.mulesoft.als.logger.Logger
...
   override def validateWithProfile(profile: String, data: String): Future[ValidationResult] = {
-    logger.debug("Starting custom validation", "JsCustomValidator", "validateWithProfile")
+    val uuid = UUID.randomUUID().toString
+    logger.debug(s"Starting custom validation $uuid", "JsCustomValidator", "validateWithProfile")
     val cb = new Callback[ValidationResult]
     AmfWasmOpaValidator.validate(profile, data, debug = false, cb.callback)
-    cb.future
+    cb.future.map(
+      r => {
+        logger.debug(s"Finished custom validation $uuid", "JsCustomValidator", "validateWithProfile")
```
Comment (llibarona) [inline als-server/js/src/main/scala/org/mulesoft/als/server/modules/diagnostic/JsCustomValidator.scala:20]:
```
    cb.future.andThen {
      case Failure(exception) => logger.error("","","")
      case Success(value) =>
        logger.debug("","","")
    }
```
    
not possible?

File: als-server/shared/src/main/scala/org/mulesoft/als/server/modules/diagnostic/custom/CustomValidationManager.scala [hunk 2]
```diff
@@ -40,6 +40,7 @@ class CustomValidationManager(override protected val telemetryProvider: Telemetr
                                      references: Map[String, DiagnosticsBundle],
                                      uuid: String): Future[Unit] = {
     val startTime = System.currentTimeMillis()
+    logger.debug(s"Starting custom validation for $uri", "CustomValidationManager", "3")
```
Comment (llibarona) [inline als-server/shared/src/main/scala/org/mulesoft/als/server/modules/diagnostic/custom/CustomValidationManager.scala:43]: should this be a telemetered message?
what is `"3"`?

File: als-server/shared/src/main/scala/org/mulesoft/als/server/modules/diagnostic/custom/CustomValidationManager.scala [hunk 3]
```diff
@@ -75,6 +76,8 @@ class CustomValidationManager(override protected val telemetryProvider: Telemetr
                                   serializedUnit: String,
                                   amfConfiguration: AmfConfigurationWrapper): Future[AMFValidationReport] =
     for {
+      _ <- Future(
+        logger.debug(s"Validate with profile: $profileUri", "CustomValidationManager", "validateWithProfile"))
```
Comment (llibarona) [inline als-server/shared/src/main/scala/org/mulesoft/als/server/modules/diagnostic/custom/CustomValidationManager.scala:80]: weird

File: als-server/js/src/main/scala/org/mulesoft/als/server/modules/diagnostic/JsCustomValidator.scala (hunk 1 above)
Comment (agusayerza) [reply als-server/js/src/main/scala/org/mulesoft/als/server/modules/diagnostic/JsCustomValidator.scala:20]: We could,  but the callback (cb) is already handling the fail case

https://github.com/aml-org/als/blob/2891a8cb0fdf0aa123bfa56a40a1bce3f6c47852/als-server/js/src/main/scala/org/mulesoft/als/server/wasm/WasmLoader.scala#L19-L27

File: als-server/shared/src/main/scala/org/mulesoft/als/server/modules/diagnostic/custom/CustomValidationManager.scala (hunk 3 above)
Comment (agusayerza) [reply als-server/shared/src/main/scala/org/mulesoft/als/server/modules/diagnostic/custom/CustomValidationManager.scala:80]: Indeed

File: als-server/shared/src/main/scala/org/mulesoft/als/server/modules/diagnostic/custom/CustomValidationManager.scala (hunk 2 above)
Comment (agusayerza) [reply als-server/shared/src/main/scala/org/mulesoft/als/server/modules/diagnostic/custom/CustomValidationManager.scala:43]: Fail xD

Comment (agusayerza) [reply als-server/shared/src/main/scala/org/mulesoft/als/server/modules/diagnostic/custom/CustomValidationManager.scala]: It already is a telemetered message, so im removing this line.

**Review by llibarona — APPROVED** [review]

**Review by llibarona — APPROVED** [review]

**Pull request merged**